OpenRA.Mods.Common/Traits/Explodes.cs

```diff
@@ -84,16 +84,30 @@ public class Explodes : ConditionalTrait<ExplodesInfo>, INotifyKilled, INotifyDa
 	{
 		readonly IHealth health;
 		BuildingInfo buildingInfo;
+		Armament[] armaments;
+
+		readonly int damageThreshold;
+		bool anyArmaments;
 
 		public Explodes(ExplodesInfo info, Actor self)
 			: base(info)
 		{
 			health = self.Trait<IHealth>();
+			damageThreshold = info.DamageThreshold;
 		}
 
 		protected override void Created(Actor self)
 		{
 			buildingInfo = self.Info.TraitInfoOrDefault<BuildingInfo>();
+			armaments = self.TraitsImplementing<Armament>().ToArray();
+
+			// PERF: Avoid LINQ
+			foreach (var a in armaments)
+			{
+				anyArmaments = true;
+				break;
+			}
```

Comment:
```suggestion
          anyArmaments = armaments.Count > 0;
			
```

Reply:
This breaks compile, does this need some kind using, or should it be `.Count()`?
Or should I just drop this?

Reply:
Sorry, should have been `armaments.Length > 0`